### DISCUSSION TITLE
Drop injecting trailing slash during sidebar generation

### DISCUSSION THREAD

**Discussion by alexander-bzikadze** [withastro/starlight#1037] [Feature Requests]
### What version of `starlight` are you using?

0.11.2

### What is your idea?

Current implementation is [injecting](https://github.com/withastro/starlight/blob/main/packages/starlight/utils/navigation.ts#L121C10-L121C41) trailing slashes into sidebar links.
This is done while ignoring [trailing slash](https://docs.astro.build/en/reference/configuration-reference/#trailingslash) option of Astro itself.

I would like to suggest either to make it configurable on the Starlight integration level or synchronized with Astro configuration.

### Why is this feature necessary?

Current implementation might prevent starlight integration into existing Astro apps, if one has been designed to work with "trailing": "never" option in mind. The effect is traceable up to deployment step with [some vendors](https://firebase.google.com/docs/hosting/full-config#control_trailing_slashes).

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
I agree that we should respect the user's `trailingSlash` config here and not add the slash if they set that to `'never'`.

@kevinzunigacuellar is currently working on a related issue in #1023 to respect `build.format` so we can probably follow up that PR with a new PR for `trailingSlash`.

**Comment by delucis** [comment]
Closing as resolved in #1023!
